File: practice/coding-cup/Card.cpp
class Card
{
private:
    int value;
    Suit suit;

public:
    enum Suit
    {
        CLUBS, SPADES, HEARTS, DIAMONDS
    };

    Card(int v, Suit s) : value(v), suit(s) {}

    int value()
    {
        return value;
    }

    Card::Suit suit()
    {
        return suit;
    }
}

File: games/oath/src/engine/card.rs
use crate::engine::types::Suit;

struct Cost {
    favor: u8,
    secrets: u8,
    burn_secrets: u8,
    burn_favor: u8,
}

enum CardRestrictionType {
    AdvisorOnly,
    SiteOnly,
}

struct CardRestriction {
    restriction: CardRestrictionType,
    locked: bool,
}

pub struct Card {
    name: String,
    suit: Suit,
    restriction: Option<CardRestriction>,
    description: String,
    cost: Cost,
}
